src/ts/ConvivaAnalytics.ts

```diff
@@ -687,25 +700,21 @@ export class ConvivaAnalytics {
     playerEvents.add(this.events.Seeked, this.onSeeked);
     playerEvents.add(this.events.TimeShift, this.onTimeShift);
     playerEvents.add(this.events.TimeShifted, this.onTimeShifted);
+
+    // Ad tracking events
+    playerEvents.add(this.events.AdStarted, this.onAdStarted);
+    playerEvents.add(this.events.AdFinished, this.onAdFinished);
   }
 
-  private onAdBreakStarted = (event: AdBreakEvent) => {
-    // Specific post-roll handling
-    const isPostRollAdBreak = (adBreak: AdBreak) => {
-      return adBreak.scheduleTime === Infinity;
-    };
+  private onAdStarted = (event: AdEvent) => {
+    this.adTrackingModule.adStarted(event);
+  };
 
-    if (isPostRollAdBreak(event.adBreak)) {
-      // Fire playbackFinished in case of a post-roll ad to stop session and do not track post roll ad
-      this.debugLog('[ ConvivaAnalytics ] detected post-roll ad ... Ending session');
-      this.onPlaybackFinished({
-          timestamp: Date.now(),
-          type: this.events.PlaybackFinished,
-        },
-      );
-      return;
-    }
+  private onAdFinished = (event: AdEvent) => {
+    this.adTrackingModule.adFinished();
+  };
 
+  private onAdBreakStarted = (event: AdBreakEvent) => {
```

Comment:
can't we initialize the session now on the adbreak started?

Reply:
Hm what do you mean actually?

This is still needed as the event order is incorrect as the comment is pointing out.

Reply:
Nevermind.